### PR TITLE
fix(openai-compat): improve pool fallback and preserve adaptive thinking

### PR DESCRIPTION
**Pull request by shenshuoyaoyouguang**
## Summary
- stop invalid pool retries in openai-compat auth conductor paths
- preserve adaptive thinking behavior when applying user-defined thinking settings
- add focused pool and thinking regression coverage

## Business value
- reduces wasted retries against invalid pools
- restores expected thinking behavior for affected requests
- keeps pool/thinking behavior separate from executor streaming fixes

## Tests
- go test ./internal/thinking ./sdk/cliproxy/auth

### CODE REVIEW

File: sdk/cliproxy/auth/conductor.go [hunk 1]
```diff
@@ -251,16 +255,309 @@ func (m *Manager) lookupAPIKeyUpstreamModel(authID, requestedModel string) strin
...
+	return strings.EqualFold(strings.TrimSpace(kind), "api_key")
+}
+
+func isOpenAICompatAPIKeyAuth(auth *Auth) bool {
+	if !isAPIKeyAuth(auth) {
+		return false
+	}
+	if auth == nil {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (shenshuoyaoyouguang) [reply sdk/cliproxy/auth/conductor.go]: Fixed in a02eda54. I removed the redundant `auth == nil` check since `isAPIKeyAuth(auth)` already guards that path.

File: sdk/cliproxy/auth/openai_compat_pool_test.go [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+	if err == nil {
+		t.Fatal("expected invalid request error")
+	}
+	if err != invalidErr {
+		t.Fatalf("error = %v, want %v", err, invalidErr)
+	}
+	if got := executor.ExecuteModels(); len(got) != 1 || got[0] != "qwen3.5-plus" {
+		t.Fatalf("execute calls = %v, want only first upstream model", got)
+	}
+}
```
[the remark this answers is not in the document]
Comment (shenshuoyaoyouguang) [reply sdk/cliproxy/auth/openai_compat_pool_test.go]: Fixed in a02eda54. I removed the duplicate invalid-request pool test and kept the earlier equivalent coverage to avoid maintaining the same scenario twice.

File: sdk/cliproxy/auth/conductor.go [hunk 3]
```diff
@@ -251,16 +255,309 @@ func (m *Manager) lookupAPIKeyUpstreamModel(authID, requestedModel string) strin
...
+			select {
+			case <-ctx.Done():
+				return nil, false, ctx.Err()
+			case chunk, ok = <-ch:
+			}
+		} else {
+			chunk, ok = <-ch
+		}
+		if !ok {
+			return buffered, true, nil
```
[the remark this answers is not in the document]
Comment (shenshuoyaoyouguang) [reply sdk/cliproxy/auth/conductor.go:425]: Fixed in a follow-up patch. An empty bootstrap stream is now treated as a retryable pooled failure instead of a successful stream, so the manager falls back to the next upstream model when the first stream closes before emitting any payload. I also added a regression test covering the empty-bootstrap pool case.

**Comment by shenshuoyaoyouguang** [comment]
Review Navigation: This PR has been supplemented with an additional round of stream pool lifecycle fixes, aiming to eliminate goroutine leak risks after context cancellation. Current CI has passed; compared to previous ones, this implementation is more complex but has returned to a reviewable state.

**Review by luispater — APPROVED** [review]
